packages/lib/src/types/primitiveBased/enum.ts

```diff
@@ -19,7 +19,8 @@ function enumValues(e: EnumLike): (string | number)[] {
     // we have to do this since TS does something weird
     // to number values
     // Hi = 0 -> { Hi: 0, 0: "Hi" }
-    if (typeof v !== "string" || e[v] !== +k) {
+    // and SWC does something weird, too
+    if (!vals.includes(v) && ((typeof v !== "string" && v !== +k) || e[v] !== +k)) {
```

Comment:
maybe this function should be unit tested with string enums, number enums and mixed number/string enums?

Reply:
Agreed, see [583aa10](https://github.com/xaviergonz/mobx-keystone/pull/411/commits/583aa10c98370ccebeb7fc33032680fc8ae17562).